ide/xml.xam/src/org/netbeans/modules/xml/xam/AbstractModel.java

```diff
@@ -179,12 +180,22 @@ public State getState() {
         return status;
     }
     
+    @Override
+    public String getStatusMessage() {
+        return statusMessage;
+    }
+    
     protected void setState(State s) {
+        setState(s, null);
+    }
+
+    protected void setState(State s, String message) {
         if (s == status) {
             return;
         }
         State old = status;
         status = s;
+        statusMessage = message;
         PropertyChangeEvent event =
```

Comment:
`statusMessage`is a different property, so I would expect both `PropertyChangeEvents` to be fired.